tests/metrics/classification/test_iou.py

```diff
@@ -0,0 +1,214 @@
+    sk_preds = (preds.view(-1).numpy() >= THRESHOLD).astype(np.uint8)
+    sk_target = target.view(-1).numpy()
+
+    return sk_jaccard_score(y_true=sk_target, y_pred=sk_preds, average=average)
+
+
+def _binary_sk_metric(preds, target, average=None):
+    sk_preds = preds.view(-1).numpy()
+    sk_target = target.view(-1).numpy()
+
+    return sk_jaccard_score(y_true=sk_target, y_pred=sk_preds, average=average)
+
+
+def _multilabel_prob_sk_metric(preds, target, average=None):
+    sk_preds = (preds.view(-1).numpy() >= THRESHOLD).astype(np.uint8)
+    sk_target = target.view(-1).numpy()
+
+    return sk_jaccard_score(y_true=sk_target, y_pred=sk_preds, average=average)
+
+
+def _multilabel_sk_metric(preds, target, average=None):
+    sk_preds = preds.view(-1).numpy()
+    sk_target = target.view(-1).numpy()
+
+    return sk_jaccard_score(y_true=sk_target, y_pred=sk_preds, average=average)
+
+
+def _multiclass_prob_sk_metric(preds, target, average=None):
+    sk_preds = torch.argmax(preds, dim=len(preds.shape) - 1).view(-1).numpy()
+    sk_target = target.view(-1).numpy()
+
+    return sk_jaccard_score(y_true=sk_target, y_pred=sk_preds, average=average)
+
+
+def _multiclass_sk_metric(preds, target, average=None):
+    sk_preds = preds.view(-1).numpy()
+    sk_target = target.view(-1).numpy()
+
+    return sk_jaccard_score(y_true=sk_target, y_pred=sk_preds, average=average)
+
+
+def _multidim_multiclass_prob_sk_metric(preds, target, average=None):
+    sk_preds = torch.argmax(preds, dim=len(preds.shape) - 2).view(-1).numpy()
+    sk_target = target.view(-1).numpy()
+
+    return sk_jaccard_score(y_true=sk_target, y_pred=sk_preds, average=average)
+
+
+def _multidim_multiclass_sk_metric(preds, target, average=None):
+    sk_preds = preds.view(-1).numpy()
+    sk_target = target.view(-1).numpy()
+
+    return sk_jaccard_score(y_true=sk_target, y_pred=sk_preds, average=average)
+
+
+@pytest.mark.parametrize("reduction", ['elementwise_mean', 'none'])
+@pytest.mark.parametrize("preds, target, sk_metric, num_classes", [
+    (_binary_prob_inputs.preds, _binary_prob_inputs.target, _binary_prob_sk_metric, 2),
+    (_binary_inputs.preds, _binary_inputs.target, _binary_sk_metric, 2),
+    (_multilabel_prob_inputs.preds, _multilabel_prob_inputs.target, _multilabel_prob_sk_metric, 2),
+    (_multilabel_inputs.preds, _multilabel_inputs.target, _multilabel_sk_metric, 2),
+    (_multiclass_prob_inputs.preds, _multiclass_prob_inputs.target, _multiclass_prob_sk_metric, NUM_CLASSES),
+    (_multiclass_inputs.preds, _multiclass_inputs.target, _multiclass_sk_metric, NUM_CLASSES),
+    (
+        _multidim_multiclass_prob_inputs.preds,
+        _multidim_multiclass_prob_inputs.target,
+        _multidim_multiclass_prob_sk_metric,
+        NUM_CLASSES
+    ),
+    (
+        _multidim_multiclass_inputs.preds,
+        _multidim_multiclass_inputs.target,
+        _multidim_multiclass_sk_metric,
+        NUM_CLASSES
+    )
+])
+class TestConfusionMatrix(MetricTester):
```

Comment:
shall it be in a separate file? or how is it related to IoU?

Reply:
should just be renamed 
```suggestion
class TestIoU(MetricTester):
```